internal/tasks/manager.go

```diff
@@ -81,16 +101,100 @@ func (t TaskManager) Stop() {
 	})
 }
 
-func (t TaskManager) SubmitTask(taskName string, resource ResourceReference) {
+func (t TaskManager) SubmitTask(taskName string, resource ResourceReference, op string) {
+	resource.Op = op
 	t.channels[taskName] <- resource
 }
 
 func (t TaskManager) GetTask(taskName string) ResourceReference {
 	return <-t.channels[taskName]
 }
 
-func (t TaskManager) FleetTemplateRolloutCallback(orgId uuid.UUID, name *string, templateUpdated bool) {
-	t.SubmitTask(FleetTemplateRollout, ResourceReference{OrgID: orgId, Name: *name})
+func (t TaskManager) FleetUpdatedCallback(before *model.Fleet, after *model.Fleet) {
```

Comment:
what if we changed the format of the callback to something like this.

````
type SomethingCallback struct {
    Before func(fleet *model.Fleet) 
    After  func(fleet *model.Fleet) 
}
```

Then you can place the  before after calls in different sections of the call. Its a common pattern to have the resource (input) as the before and the response and the after. I think it might simplify some of this logic.

Reply:
Sorry, I don't follow

Reply:
we can talk about this more as followup